### PR TITLE
feat(ec2): ApplicationLoadBalancer support

### PR DESCRIPTION
**Pull request by asher**

### CODE REVIEW

File: keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandler.kt [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  override val objectMapper: ObjectMapper,
+  override val normalizers: List<ResourceNormalizer<*>>
+) : ResourceHandler<ApplicationLoadBalancer> {
+  override val log: Logger by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override val apiVersion = SPINNAKER_API_V1.subApi("ec2")
+  override val supportedKind = ResourceKind(
+    apiVersion.group,
+    "alb",
+    "albs"
```
Comment (robfletcher) [inline keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandler.kt]: Pedantic but would it make sense to use something more verbose here? We currently have `cluster`, `security-group` and `image`. Is `application-load-balancer` too long? I like the clarity myself.

File: keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/normalizers/ApplicationLoadBalancerNormalizer.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+) : ResourceNormalizer<ApplicationLoadBalancer> {
+  override val apiVersion = SPINNAKER_API_V1.subApi("ec2")
+  override val supportedKind = "alb"
+
+  override fun normalize(resource: Resource<*>): Resource<ApplicationLoadBalancer> {
+    @Suppress("UNCHECKED_CAST")
+    val r = resource as Resource<ApplicationLoadBalancer>
+
+    if (r.spec.targetGroups.any { it.name.length > 32 }) {
+      throw IllegalStateException("targetGroup names have a 32 character limit")
```
Comment (robfletcher) [inline keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/normalizers/ApplicationLoadBalancerNormalizer.kt]: We definitely need to think about how to model this kind of thing in the schema

File: keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandler.kt (hunk 1 above)
Comment (asher) [reply keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandler.kt]: I like keeping it pithy but I'm fine with verbose, will update `clb` to `classic-load-balancer` as well.

File: keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/normalizers/ApplicationLoadBalancerNormalizer.kt (hunk 2 above)
Comment (asher) [reply keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/normalizers/ApplicationLoadBalancerNormalizer.kt]: Looks like precondition checks can be added to a data class' init block, going to test that out.

Comment (robfletcher) [reply keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/normalizers/ApplicationLoadBalancerNormalizer.kt]: I wasn't suggesting you need to do it as part of this PR, just thinking aloud.

**Pull request merged**